src/main/java/org/folio/services/financedata/FinanceDataService.java

```diff
@@ -123,14 +129,18 @@ private Future<FyFinanceDataCollection> updateFinanceData(FyFinanceDataCollectio
 
   private Future<FundUpdateLog> processLogs(String fundUpdateLogId, FyFinanceDataCollection financeDataCollection,
                                             RequestContext requestContext) {
+    var ledgerId = Optional.ofNullable(financeDataCollection.getFyFinanceData().getFirst().getLedgerId())
+      .orElseThrow(() -> new NoSuchElementException("Ledger id is not found"));
```

Comment:
if ledgerId is required one I think we need to add it to validation as required field https://github.com/folio-org/mod-finance/blob/a20d20d8618d7c04df55b46c6549884f9a0ff549/src/main/java/org/folio/services/financedata/FinanceDataValidator.java#L89

Reply:
and there will be just get without throw